### PR TITLE
slurm: init at 0.4.4

### LINKED ISSUE #257288: Package request: slurm
**Project description**

> yet another network load monitor

slurm started as a FreeBSD port of the Linux ppp link monitor called pppstatus by Gabriel Montenegro. Hendrik Scholz ripped off the ppp dependent parts and the email checks to turn in into a generic network load monitor for *BSD, Linux, HP-UX and Solaris.

**Metadata**

* homepage URL: https://github.com/mattthias/slurm
* source URL: https://github.com/mattthias/slurm
* license: gpl2
* platforms: unix, linux

### PR DESCRIPTION
**Pull request by MikaelFangel**
## Description of changes

Closes #257288

Adds the package slurm as a generic network load monitor. The name of the package is slurm-cli as there already is a package named slurm.

URL: [https://github.com/mattthias/slurm](https://github.com/mattthias/slurm)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SebTM** [comment]
You are missing the "include" of the new package in `pkgs/top-level/all-packages.nix` like:
```
  slurm-cli = callPackage ../tools/networking/slurm-cli { };
```

File: pkgs/tools/networking/slurm-cli/default.nix
Comment (eclairevoyant) [inline pkgs/tools/networking/slurm-cli/default.nix:31]: gpl2 is deprecated, always use gpl2Plus or gpl2Only depending on the repo licensing.
in this case, they explicitly allow "any later version" in the source code headers (e.g. look at slurm.c), so:
```suggestion
    license = licenses.gpl2Plus;
```

File: pkgs/tools/networking/slurm-cli/default.nix
Comment (eclairevoyant) [inline pkgs/tools/networking/slurm-cli/default.nix:26]: Normally I'd say, please keep the phase hooks as mentioned here when overriding a phase: https://nix.dev/tutorials/learning-journey/packaging-existing-software#phases-and-hooks

But this code is not doing anything, best remove it.

```suggestion
```

File: pkgs/tools/networking/slurm-cli/default.nix
Comment (eclairevoyant) [inline pkgs/tools/networking/slurm-cli/default.nix:17]:
```suggestion
    rev = "upstream/${version}";
```

File: pkgs/tools/networking/slurm-cli/default.nix
Comment (eclairevoyant) [inline pkgs/tools/networking/slurm-cli/default.nix:33]:
```suggestion
    maintainers = [ maintainers.mikaelfangel ];
    mainProgram = "slurm";
```

`meta.mainProgram` should be set to the name of the main binary, when there is one

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Thanks for the contribution!

As per the new process, please move this file to the new by-name hierarchy:
`pkgs/by-name/sl/slurm-cli/package.nix`

Also in the future, please test the PRs before opening, it makes the review process much smoother :) You can do this, in this case, by running `nix-build -A slurm-cli` in the root of your clone.

Finally, we would need an additional commit prior to your current one, adding yourself as a maintainer (see https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions):

> When adding yourself as maintainer, in the same pull request, make a separate commit with the message `maintainers: add <handle>`. Add the commit before those making changes to the package or module. See [Nixpkgs Maintainers](https://github.com/NixOS/nixpkgs/blob/maintainers/README.md) for details.

**Comment by eclairevoyant** [comment]
(Also out of curiosity, any reason to name it `slurm-cli` instead of `slurm`?)

**Comment by tobim** [comment]
Just guessing: To avoid a conflict with `pkgs/servers/computing/slurm`.

**Comment by MikaelFangel** [comment]
As @tobim said and because i thought that slurm-network-monitor would be a bit long. Any suggestions for another name @eclairevoyant ?

**Comment by eclairevoyant** [comment]
Makes sense. I don't have a strong suggestion about the naming, as I'm not familiar with the software. Just wanted to check.

I'll approve once you get a chance to move the file and the commits are created per the guidelines (1 maintainers commit, 1 commit to create this new package)

**Comment by SebTM** [comment]
I personally (when thinking about it) would prefer slurm-nm/netmon or sth because with cli like e.g. 1password, 1password-cli someone could think packages are related

**Comment by MikaelFangel** [comment]
Cool, I'm working on it now and try to correct the errors from the checks (:

**Comment by MikaelFangel** [comment]
@eclairevoyant thank you for thanking the time and reviewing my PR despite all my clear oversights in the contribution guidelines. I've learned a lot! (:

### Changes:
- [ ] Moved to by-name hierarchy
- [ ] Test built it with `nix-build -A`
- [ ] Renamed the package to slurm-nm
- [ ] Added my-self as a maintainer
- [ ] Updated the meta
- [ ] Applied suggestions from code review

However, it says that the maintainer list is not sorted, but maybe I misunderstand something. But, mikael comes after mihnea and before mike, right?

**Review by eclairevoyant — APPROVED** [review]

File: pkgs/by-name/sl/slurm-nm/package.nix
Comment (Janik-Haag) [inline pkgs/by-name/sl/slurm-nm/package.nix:24]:
```suggestion
```
just a really small formatting nitpick looks good to me except for that

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 257357` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>slurm-nm</li>
  </ul>
</details>

**Review by Janik-Haag — APPROVED** [review]

**Pull request merged**
